executor/change.go

```diff
@@ -0,0 +1,62 @@
+// Copyright 2019 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package executor
+
+import (
+	"context"
+	"strings"
+
+	"github.com/pingcap/errors"
+	"github.com/pingcap/parser/ast"
+	"github.com/pingcap/tidb-tools/tidb-binlog/node"
+	"github.com/pingcap/tidb/config"
+	"github.com/pingcap/tidb/util/chunk"
+)
+
+// ChangeExec represents a change executor.
+type ChangeExec struct {
+	baseExecutor
+
+	Statement *ast.ChangeStmt
+}
+
+// Next implements the Executor Next interface.
+func (e *ChangeExec) Next(ctx context.Context, req *chunk.RecordBatch) error {
+	stmt := e.Statement
+	kind := strings.ToLower(stmt.NodeType)
+	urls := config.GetGlobalConfig().Path
+	registry, err := createRegistry(urls)
+	if err != nil {
+		return err
+	}
+	nodes, _, err := registry.Nodes(ctx, node.NodePrefix[kind])
+	if err != nil {
+		return err
+	}
+	state := stmt.State
+	nodeID := stmt.NodeID
+	for _, n := range nodes {
+		if n.NodeID != nodeID {
+			continue
+		}
+		switch state {
+		case node.Online, node.Pausing, node.Paused, node.Closing, node.Offline:
+			n.State = state
+			return registry.UpdateNode(ctx, node.NodePrefix[kind], n)
```

Comment:
Should we compare the old state and new state? Thus we can save the network connection if nothing changed.

Reply:
if want to know the old state, need send a request to the pd, and then get the status information, this can't save the network. 

and this operate only do in some special situation, may execute one sql several days.